internal/cron/verify_domain_ownership.go

```diff
@@ -0,0 +1,88 @@
+package cron
+
+import (
+	"context"
+	"errors"
+	"net"
+
+	"github.com/oursky/pageship/internal/db"
+	"github.com/oursky/pageship/internal/models"
+	"github.com/oursky/pageship/internal/time"
+	"go.uber.org/zap"
+)
+
+type VerifyDomainOwnership struct {
+	Schedule                     string
+	DB                           db.DB
+	MaxConsumeActiveDomainCount  uint
+	MaxConsumePendingDomainCount uint
+}
+
+func (v *VerifyDomainOwnership) Name() string { return "verify-domain-ownership" }
+
+func (v *VerifyDomainOwnership) CronSchedule() string { return v.Schedule }
+
+func (v *VerifyDomainOwnership) Run(ctx context.Context, logger *zap.Logger) error {
+	now := time.SystemClock.Now().UTC()
+
+	return db.WithTx(ctx, v.DB, func(c db.Tx) error {
+		verified := true
+		unverified := false
+		domainVerifications := []*models.DomainVerification{}
+		verifiedRecords, _ := c.ListDomainVerifications(ctx, nil, &v.MaxConsumeActiveDomainCount, &verified)
+		unverifedRecords, _ := c.ListDomainVerifications(ctx, nil, &v.MaxConsumePendingDomainCount, &unverified)
```

Comment:
I'd prefer to have an explicit method for this purpose, something like `ListLeastRecentlyCheckedDomain(isVerified, count)`.
- The order of the listing is not (?) part of the API interface. For example, one may found ordering by domain name is more natural.
- Omitting app ID should not be possible for normal list.
- The currently listing method is too complex with many string joins.